error[E0425]: cannot find function `get_internal` in this scope when compiling to WASM

Error message reads:
```
[INFO]: Checking for the Wasm target...
[INFO]: Compiling to Wasm...
   Compiling thread-id v3.3.0
   Compiling backtrace-sys v0.1.31
   Compiling backtrace v0.3.34
error[E0425]: cannot find function `get_internal` in this scope
  --> /home/ben/.cargo/registry/src/github.com-1ecc6299db9ec823/thread-id-3.3.0/src/lib.rs:44:5
   |
44 |     get_internal()
   |     ^^^^^^^^^^^^ not found in this scope

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
error: Could not compile `thread-id`.
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit code: 101
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -65,6 +65,36 @@ fn get_internal() -> usize {
     syscall::getpid().unwrap()
 }
 
+#[cfg(all(
+    target_arch = "wasm32",
+    target_vendor = "unknown",
+    target_os = "unknown",
+    not(target_feature = "atomics"),
+))]
+#[inline]
+fn get_internal() -> usize {
+    0
+}
+
+#[cfg(all(
+    target_arch = "wasm32",
+    target_vendor = "unknown",
+    target_os = "unknown",
+    target_feature = "atomics",
+))]
+#[inline]
+fn get_internal() -> usize {
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    static COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+    thread_local! {
+        static ID: usize = COUNTER.fetch_add(1, Ordering::Relaxed);
+    }
+
+    ID.with(|id| *id)
+}
+
 #[test]
 fn distinct_threads_have_distinct_ids() {
     use std::sync::mpsc;

